test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -466,8 +466,78 @@ def get_number(field)
     field['numberValue']
   end
 
+  def get_bool(field)
+    field['boolValue']
+  end
+
   # The null value.
   def null_value
     { 'nullValue' => 'NULL_VALUE' }
   end
+
+  def http_request_message
+    HTTP_REQUEST_MESSAGE
+  end
+
+  def source_location_message
+    SOURCE_LOCATION_MESSAGE
+  end
+
+  def source_location_message2
+    SOURCE_LOCATION_MESSAGE2
+  end
+
+  def expected_operation_message2
+    # 'last' is a boolean field with false as default value. Protobuf omit
+    # fields with default values during deserialization.
+    OPERATION_MESSAGE2.reject { |k, _| k == 'last' }
+  end
+
+  # expected: A Ruby hash that represent a JSON object.
+  # e.g.:
+  #   {
+  #     "file" => "source/file",
+  #     "function" => "my_function",
+  #     "line" => 18
+  #   }
+  #
+  # actual: A Ruby hash that represents a Proto object.
+  # e.g.:
+  #   {
+  #     "structValue" => {
+  #       "fields" => {
+  #         "file" => {
+  #           "stringValue" => "source/file"
+  #         },
+  #         "function" => {
+  #           "stringValue" => "my_function"
+  #         },
+  #         "line" => {
+  #           "numberValue" => 18
+  #         }
+  #       }
+  #     }
+  #   }
+  def assert_json_equal(expected, actual)
```

Comment:
assert_json_equal_proto highlights the fact that these are different structures.

Reply:
I thought about this, then realized this method name has to be shared between gRPC and REST. The REST side does not involve proto. I just added some comments for clarification.

Reply:
How about hash_equal_json or something where the reader understands at the call location, without having to read comments?

Reply:
Just renamed this to `assert_hash_equal_json`. WDYT?